### ISSUE TITLE
Add check when adding BrewPiDevices for uniqueness of device_name

### ISSUE THREAD

**Issue by thorrak** [thorrak/fermentrack#273]
Device names currently are required to be unique by the database, but this uniqueness isn't checked when the form is cleaned. This needs to be added, so the form can be properly invalidated (and a valid error message presented to the user)

```
IntegrityError: UNIQUE constraint failed: app_brewpidevice.device_name
  File "django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "django/db/backends/sqlite3/base.py", line 328, in execute
    return Database.Cursor.execute(self, query, params)

IntegrityError: UNIQUE constraint failed: app_brewpidevice.device_name
(15 additional frame(s) were not displayed)
...
  File "django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "django/db/utils.py", line 94, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "django/utils/six.py", line 685, in reraise
    raise value.with_traceback(tb)
  File "django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "django/db/backends/sqlite3/base.py", line 328, in execute
    return Database.Cursor.execute(self, query, params)

IntegrityError: UNIQUE constraint failed: app_brewpidevice.device_name
```

**Comment by mp-se** [comment]
This issue seams to be fixed (at least for the advanced add)

**Comment by thorrak** [comment]
This issue is partially fixed now, but a complete fix is coming with the fix for #355 . Closing this one now - the other issue will get closed when I merge the fix in.
